### PR TITLE
ss-path attribute

### PR DESCRIPTION
**Pull request by chrisFrazier77**

### CODE REVIEW

**Comment by chrisFrazier77** [comment]
There is something interesting happening with the treepath prop being spread onto dropdown buttons. you can see this on all of the facet headers. its not the expected 'ss-path' attribute we are creating, but rather the treepath prop being spread on as an attribute. Even though the button component is pulling that prop out before spreading "additionalProps". I believe it has something to do with cloneWithProps, since we are passing the treepath into those props.

File: packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleCTA.tsx
Comment (korgon) [inline packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleCTA.tsx:1]: Additionally, shouldn't the `ss-path` here include the icon name?
image

File: packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleCTA.tsx [hunk 1]
```diff
@@ -32,37 +35,42 @@ export const BundledCTA = observer((properties: BundledCTAProps): JSX.Element =>
 
 	const { ctaSlot, cartStore, onAddToCart, ctaIcon, ctaButtonText, ctaButtonSuccessText, treePath } = props;
 
+	const lastPath = treePath?.lastIndexOf(' ');
+	const modifiedTreePath = treePath?.slice(0, lastPath);
```
Comment (korgon) [inline packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleCTA.tsx:39]: I'm guessing that the `mergeProps()` usage is causing the need for this? Related to a previous comment, what is the purpose of using mergeProps here?

File: packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleSelector.tsx [hunk 2]
```diff
@@ -20,16 +20,20 @@ export const BundleSelector = observer((properties: BundleSelectorProps): JSX.El
 
 	const { children, checked, icon, seedText, seed, hideCheckboxes, onCheck, title, className, treePath } = props;
 
+	const lastPath = treePath?.lastIndexOf(' ');
+	const modifiedTreePath = treePath?.slice(0, lastPath);
+
 	const subProps: BundleSelectorSubProps = {
 		icon: {
 			// default props
+			name: 'bundle-selector-icon',
```
Comment (korgon) [inline packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleSelector.tsx:29]: Maybe better named as `bundle-separator` or `bundle-selector-separator` ?

File: packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleSelector.tsx
Comment (korgon) [inline packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleSelector.tsx:1]: The name is not showing up in the `ss-path` here as expected...
image

Comment (korgon) [inline packages/snap-preact/components/src/components/Templates/RecommendationBundle/BundleSelector.tsx:1]: Might be a good idea to give this component a name, as there is a potential to have other checkboxes within the product cart (think custom card) making it hard to target this one specifically...  🤷🏻‍♂️ 
image

**Review by korgon — CHANGES_REQUESTED** [review]
The `ss-path` should make it easier to identify for CCP usage.

I was unable to test selecting via names due to the bug:
https://github.com/searchspring/snap/issues/1211

**Review by korgon — APPROVED** [review]
Working well. I think this is now in a good spot.

**Pull request merged**